torch/autograd/gradcheck.py

```diff
@@ -53,38 +62,47 @@ def get_numerical_jacobian(fn, input, target=None, eps=1e-3):
     if target is None:
         target = input
     output_size = fn(input).numel()
-    jacobian = make_jacobian(target, output_size)
-
+    jacobian = make_jacobian(target, output_size, extra_dim_if_complex=True)
     # It's much easier to iterate over flattened lists of tensors.
     # These are reference to the same objects in jacobian, so any changes
     # will be reflected in it as well.
     x_tensors = iter_tensors(target, True)
     j_tensors = iter_tensors(jacobian)
 
-    def compute_gradient(x, idx, is_mkldnn=False):
-
-        def fn_out():
-            if not is_mkldnn:
-                # x is a view into input and so this works
-                return fn(input).clone()
-            else:
-                # convert the dense tensor back to have mkldnn layout
-                return fn([x.to_mkldnn()])
-
-        orig = x[idx].item()
-        x[idx] = orig - eps
-        outa = fn_out()
-        x[idx] = orig + eps
-        outb = fn_out()
-        x[idx] = orig
-        r = (outb - outa) / (2 * eps)
-        return r.detach().reshape(-1)
+    def update_jacobians(x, idx, d, d_idx, is_mkldnn=False):
+
+        def compute_gradient(delta=eps):
```

Comment:
Do you need a default value here? You always provide it below.

Reply:
yup it's unnecessary. will remove it